### PR TITLE
To do list: Add edit and remove todo

### PR DESCRIPTION
**Pull request by ernestmusong**
## I have added the following functionalities to my TO DO LIST:
-**Display todos:**  Todos are fetched and displayed from local storage as page loads.
- **Add To do:** A user can add a to-do to the list and it's saved to local storage.
- **Remove To do:** A user can remove a to-do from the list and the local storage gets updated.
- **Edit To do:** A user can edit or update a to-do and the local storage gets updated.
- The number of items in the list is displayed on a counter at the top-right of the application.
**Every to has a dynamic index value assigned to a todo item.**

You can view the [Live domo here](https://heroic-cupcake-639ac3.netlify.app/)

### CODE REVIEW

File: src/ui.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    const deleteBtns = [...document.querySelectorAll('.drag')];
+    deleteBtns.forEach((btn) => {
+      if (e.target === btn) {
+        const filtered = todoArr.filter((item) => item.index.toString() !== e.target.id.toString());
+        todoArr = filtered;
+        localStorage.setItem('todos', JSON.stringify(todoArr));
+        UI.displayTodos();
+      }
+    });
+  }
```
Comment (amejid) [inline src/ui.js]: - [ ] Great job implementing all the core functionalities. :+1: Let's work on some minor detail to achieve the objective of this project. Kindly update indexes of the remaining items when a to-do is removed. ✍️

**Project Requirements**
Screenshot from 2023-02-24 11-40-09

File: src/ui.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      for (let i = 0; i < todoArr.length; i += 1) {
+        todoArr[i].index = i + 1;
+      }
+      localStorage.setItem('todos', JSON.stringify(todoArr));
+    }
+  }
+
+  static displayTodos() {
+    const todos = JSON.parse(localStorage.getItem('todos'));
+    counter.textContent = todos.length;
```
Comment (amejid) [inline src/ui.js:26]: - I like how you are displaying the list count as a badge. :100: Nice work.

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @ernestmusong, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- ADD, EDIT, REMOVE functionalities working as expected :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/ui.js (hunk 1 above)
Comment (ernestmusong) [reply src/ui.js]: > * [ ]  Great job implementing all the core functionalities. 👍 Let's work on some minor detail to achieve the objective of this project. Kindly update indexes of the remaining items when a to-do is removed. ✍️
> 
> **Project Requirements** Screenshot from 2023-02-24 11-40-09

Hi @amejid,
 Thanks for your feedback. I have  now implemented the above requirement.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @ernestmusong 👋, 

Well done for implementing the required changes 👏 
Your project is now complete!  And There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**